src/core/views.py

```diff
@@ -2420,9 +2428,22 @@ def get_queryset(self, params_querydict=None):
             # The following line prevents the user from passing any parameters
             # other than those specified in the facets.
             if keyword in facets and value_list:
-                if value_list[0]:
+                if facets[keyword]['type'] == 'search':
+                    self.queryset, _duplicates = search_model_admin(
+                        self.request,
+                        self.model,
+                        q=value_list[0],
+                        queryset=self.queryset,
+                    )
+                    predicates = []
+                elif facets[keyword]['type'] == 'boolean':
+                    if value_list[0]:
+                        predicates = [(keyword, True)]
+                    else:
+                        predicates = [(keyword, False)]
+                elif value_list[0]:
                     predicates = [(keyword, value) for value in value_list]
-                elif facets[keyword]['type'] not in ['date_time', 'date']:
+                elif facets[keyword]['type'] not in ['date_time', 'date', 'integer', 'search']:
```

Comment:
Thanks for pushing on this piece. I did some digging to try to remember *exactly* why it's there, and I think it is actually not needed, because we used a different method in the end. I'll document this in the commit message body removing this if/else check.